src/queryResult/queryResultWebViewController.ts

```diff
@@ -78,17 +89,104 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
             state.tabStates.resultPaneTab = payload.tabId;
             return state;
         });
+        this.registerReducer("getExecutionPlan", async (state, payload) => {
+            await this.createExecutionPlanGraphs(state, payload.xmlPlans);
+            state.executionPlanState.loadState = ApiStatus.Loaded;
+            state.tabStates.resultPaneTab =
+                qr.QueryResultPaneTabs.ExecutionPlan;
+            return state;
+        });
+        this.registerReducer("addXmlPlan", async (state, payload) => {
+            state.executionPlanState.xmlPlans = [
+                ...state.executionPlanState.xmlPlans,
+                payload.xmlPlan,
+            ];
+            return state;
+        });
+        this.registerReducer("saveExecutionPlan", async (state, payload) => {
+            let folder = vscode.Uri.file(homedir());
+            let filename: vscode.Uri;
+
+            // make the default filename of the plan to be saved-
+            // start ad plan.sqlplan, then plan1.sqlplan, ...
+            let counter = 1;
```

Comment:
Is this the same logic as in executionPlanWebviewController?  Can you pull it into a shared helper/utils method?  Something like

`getUniquePath(fileBaseName: string, extension: string, path: Uri): Uri`

Reply:
Tracking in #18186